backends/lean/Base/Progress/Progress.lean

```diff
@@ -370,14 +373,22 @@ def evalProgress (args : TSyntax `Progress.progressArgs) : TacticM Unit := do
       Arith.scalarTac false
     else
       throwError "Not a linear arithmetic goal"
-  progressAsmsOrLookupTheorem keep withArg ids splitPost (
+  let usedTheorem ← progressAsmsOrLookupTheorem keep withArg ids splitPost (
     withMainContext do
     trace[Progress] "trying to solve assumption: {← getMainGoal}"
     firstTac [assumptionTac, scalarTac])
-  trace[Diverge] "Progress done"
```

Comment:
@sonmarcho here, I think it was meant to be Progress, I renamed it even though it has nothing to do with this PR, is this fine or should I split the change?

Reply:
Yes, that's fine, thanks! :)